addon/session-stores/cookie.js

```diff
@@ -189,7 +189,7 @@ export default BaseStore.extend({
       document.cookie = `${this._oldCookieName}=`;
     }
 
-    if (isBlank(value)) {
+    if (isBlank(value) && expiration !== 0) {
```

Comment:
I don't think this is correct. If the expiration is a number other than 0 you'd still want to write the cookie - it would just expire after `expiration` seconds.